.github/workflows/release.yml

```diff
@@ -58,18 +64,40 @@ jobs:
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4.2.2
-      - name: Mixed Mode Test
-        uses: ./actions/mixed-mode-test
+      - name: Setup Base Environment
+        uses: ./actions/setup-base-env
+      - name: Setup FDB
+        uses: ./actions/setup-fdb
+      - name: Run Gradle Test
+        uses: ./actions/gradle-test
         with:
-          update_type: ${{ needs.get-update-type.outputs.update-type }}
+          gradle_command: mixedModeTest
+          gradle_args: -PreleaseBuild=false -PpublishBuild=false
+      # We don't commit the incremented version, but we use this to know the version when generating
+      # the resulting markdown
+      - name: Increment version
+        shell: bash
+        run: python build/versionutils.py gradle.properties --increment -u ${{ needs.get-update-type.outputs.update-type }}
+      - name: Get new version
+        id: get_new_version
+        shell: bash
+        run: |
+          echo "version=$(python build/versionutils.py gradle.properties)" >> "$GITHUB_OUTPUT"
+      - name: Create markdown
+        shell: bash
+        run: python build/publish-mixed-mode-results.py ${{ steps.get_new_version.outputs.version }} --run-link ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }} --output mixed-mode-results.md
+      - name: Preview results
+        shell: bash
+        run: cat mixed-mode-results.md >> $GITHUB_STEP_SUMMARY
       # I think this _needs_ to be done at this level, rather than in the action
       # so that "mixed-mode-results" gets passed around correctly
       - name: Upload mixed mode results
         id: mixed_mode_results
-        uses: actions/upload-pages-artifact@v3
+        uses: actions/upload-artifact@v4.6.0
         with:
           name: mixed-mode-results
-          path: ./artifact
+          path: mixed-mode-results.md
```

Comment:
We should consider adding `mixed-mode-results.md` to the `.gitignore`, or placing it in a directory that is already excluded. The way the PR creating workflow works, if there are files that aren't in the `.gitignore` present, it will include them in the PR that it creates. Excluding the file would make sure that we don't accidentally do that, particularly in the other step when we download it.